docs/diagram.md

```diff
@@ -38,4 +38,12 @@ The agent operation breaks down into three parts:
     * Multiplexer (Thread 4 - Asynchronous event loop): the multiplexer has an event loop which processes incoming and outgoing messages across several connections asynchronously.
 * Teardown: calls the `teardown()` method of all registered resources
 
-<br />
\ No newline at end of file
+
+To avoid block of main loop execution time limit applied to every `Behaviour.act` and `Handler.handle`
+
+Default execution limit is 1 second. If act or handle time exceed limit, code will be terminated.
+
+Appropriate message will be added to logs in case of some code execution was terminated.
```

Comment:
```suggestion
To prevent a developer from blocking the main loop with custom skill code, an execution time limit is  applied to every `Behaviour.act` and `Handler.handle` call.

The default execution limit is `1` second. If the `act` or `handle` time exceed this limit, the call will be terminated.

An appropriate message is added to the logs in the case of some code execution being terminated.
```